substrate/frame/tips/src/lib.rs

```diff
@@ -611,4 +623,25 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Tips::<T, I>::insert(hash, new_tip)
 		}
 	}
+
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// This should be valid before or after each state transition of this pallet.
+	///
+	/// ## Invariants:
+	///
+	/// If `OpenTip.finders_fee` is true, then OpenTip.deposit should be greater that zero.
+	#[cfg(any(feature = "try-runtime", test))]
+	pub fn do_try_state() -> Result<(), TryRuntimeError> {
+		for tips in Tips::<T, I>::iter_keys() {
+			let open_tip = Tips::<T, I>::get(&tips).unwrap();
+			if open_tip.finders_fee {
+				ensure!(
+					!open_tip.deposit.is_zero(),
+					TryRuntimeError::Other("finder's fee not present")
```

Comment:
I think these errors should be as descriptive as possible. For example: 
```suggestion
					TryRuntimeError::Other("Tips with `finders_fee` should have non-zero `deposit`.")
```